template/cordova/lib/MSBuildTools.js

```diff
@@ -44,8 +52,39 @@ MSBuildTools.prototype.buildProject = function(projFile, buildType, buildarch, o
             args.push('/p:' + key + '=' + otherConfigProperties[key]);
         });
     }
+    var that = this;
 
-    return spawn(path.join(this.path, 'msbuild'), [projFile].concat(args), { stdio: 'inherit' });
+    // Check if SDK required to build the respective platform is present. If not present, return with corresponding error, else call msbuild.
+    if (projFile.indexOf('CordovaApp.Phone.jsproj') > -1) {
```

Comment:
I thought about something like that (assuming that `check*SDK` will throw if SDK not found):
```
var promise;

if (projFile.indexOf('CordovaApp.Phone.jsproj') > -1) {
    promise = checkPhoneSDK();
} else if (projFile.indexOf('CordovaApp.Windows.jsproj') > -1) {
    promise = checkWinSDK('8.1');
} else {
    promise = checkWinSDK('10.0');
}

return promise.then(function () {
    return spawn(path.join(that.path, 'msbuild'), [projFile].concat(args), { stdio: 'inherit' });
})
```